### DISCUSSION TITLE
[Modeler3D]unable to sweep along path

### DISCUSSION THREAD

**Discussion by gnimdu** [ansys/pyaedt#3459] [Q&A] [answered]
I encountered a strange behaviour of the modeler.
I was trying to build part of an helix that could have different section.

I use these two approaches : 

**1.  first approach (using circle)** 

_- Section creation_
`center_pos = [0,-0.02,0]`
`dimensions = [0.005, 0.005]`
`position = [center_pos[0]-0.005/2, center_pos[1]-0.005/2, 0]`
`coil1 = modeler.create_circle(cs_plane="YZ", position=center_pos, radius=0.0025, name="coil1")`

_- Path creation_
`sweep_path_1 = modeler.create_equationbased_curve(x_t=str(-0.02)+"*sin(_t)",
                                            y_t=str(0.02)+"*cos(_t)",
                                            z_t=str(0.00087)+"*_t",
                                            t_start=0,
                                            t_end=math.pi/2,
                                            num_points=0,
                                            name="sweep_path_1",
                                            )`

_- Sweep along path_

`sweep_res1 = modeler.sweep_along_path(coil1, sweep_path_1)`

![QA_sweep0](https://github.com/ansys/pyaedt/assets/137474865/72450a65-195b-424c-8b86-b8cc1f034cb4)

**2.second approach (using rectangle+fillet)** 
_-Section _creation_
`center_pos = [0,-0.02,0]`
`dimensions = [0.005, 0.005]`
`position = [center_pos[0]-0.005/2, center_pos[1]-0.005/2, 0]`
`coil1 = modeler.create_rectangle(csPlane="YZ", position=position, dimension_list=dimensions, name="coil1")`
`coil1_vertice = coil1.vertices`
`coil1_vertice[0].fillet(0.0025)`
`coil1_vertice[1].fillet(0.0025)`
`coil1_vertice[2].fillet(0.0025)`
`coil1_vertice[3].fillet(0.0025)`

the path creation and the sweep remain the same.

Everything went fine with the first approach, but i get this error with the second approach although i was excpecting the same result.

![QA_sweep](https://github.com/ansys/pyaedt/assets/137474865/2f2df2a0-76a4-4722-a260-fea8f2a966df)

Can anybody help me understand why this last approach is not working?

**Comment by Samuelopez-ansys** [comment, marked as answer]
@gnimdu I guess that was an issue from previous AEDT versions with Parasolid kernel. I tried in 2023R2 and PyAEDT v0.7.10, and it works:

```
import pyaedt
import math

aedtapp = pyaedt.Hfss(specified_version="2023.2")

center_pos = [0,-0.02,0]
dimensions = [0.005, 0.005]
position = [center_pos[0]-0.005/2, center_pos[1]-0.005/2, 0]
coil1 = aedtapp.modeler.create_circle(cs_plane="YZ", position=center_pos, radius=0.0025, name="coil1")

sweep_path_1 = aedtapp.modeler.create_equationbased_curve(x_t=str(-0.02)+"*sin(_t)", y_t=str(0.02)+"*cos(_t)", z_t=str(0.00087)+"*_t", t_start=0, t_end=math.pi/2, num_points=0, name="sweep_path_1", )

sweep_res1 = aedtapp.modeler.sweep_along_path(coil1, sweep_path_1)


center_pos = [0,-0.02,0]
dimensions = [0.005, 0.005]
position = [center_pos[0]-0.005/2, center_pos[1]-0.005/2, 0]
coil2 = aedtapp.modeler.create_rectangle(csPlane="YZ", position=position, dimension_list=dimensions, name="coil2")
coil2_vertice = coil2.vertices
coil2_vertice[0].fillet(0.0025)
coil2_vertice[1].fillet(0.0025)
coil2_vertice[2].fillet(0.0025)
coil2_vertice[3].fillet(0.0025)

sweep_path_2 = aedtapp.modeler.create_equationbased_curve(x_t=str(-0.02)+"*sin(_t)", y_t=str(0.02)+"*cos(_t)", z_t=str(0.00087)+"*_t", t_start=0, t_end=math.pi/2, num_points=0, name="sweep_path_2", )

sweep_res2 = aedtapp.modeler.sweep_along_path(coil2, sweep_path_2)


aedtapp.release_desktop(False, False)
```
